packages/devtools_app/lib/src/screens/memory/shared/heap/model.dart

```diff
@@ -43,12 +44,20 @@ class AdaptedHeapData {
     );
   }
 
-  factory AdaptedHeapData.fromHeapSnapshot(HeapSnapshotGraph graph) =>
-      AdaptedHeapData(
-        graph.objects
-            .map((e) => AdaptedHeapObject.fromHeapSnapshotObject(e))
-            .toList(),
-      );
+  static Future<AdaptedHeapData> fromHeapSnapshot(
+    HeapSnapshotGraph graph,
+  ) async {
+    final listOfFutures = graph.objects.map((e) async {
+      await delayForBatchProcessing();
+      return AdaptedHeapObject.fromHeapSnapshotObject(e);
+    }).toList();
+
+    final objects = await Future.wait(listOfFutures);
+
+    // The future is passed as an argument intentionally.
+    // ignore: discarded_futures
```

Comment:
Is this ignore necessary? `objects` isn't an instance of `Future` at this point.